src/server/rpc/procedures/geolocation/geolocation.js

```diff
@@ -33,7 +39,7 @@ if(!process.env.GOOGLE_GEOCODING_API) {
 
     // reverse geocoding helper, doesn't return a promise. handles sending of response.
     let reverseGeocode = (lat, lon, response, query)=>{
-        cache.wrap(lat + ', ' + lon + query, cacheCallback => {
+        cache.wrap(coordsToCacheKey(lat, lon) + query, cacheCallback => {
```

Comment:
Why are you caching using the query? You should be able to cache the whole thing once across queries

Reply:
keeping the changes minimal. In addition, some parts are using are making the calls through a library and in other parts, the service manually makes the calls